src/configuration.cpp

```diff
@@ -728,13 +791,49 @@ bool chimera::CompiledConfiguration::Render(
                   << std::endl;
         return false;
     }
-    std::string mangled_name = ::mstch::render("{{mangled_name}}", context);
+    const std::string mangled_name
+        = ::mstch::render("{{mangled_name}}", context);
+
+    // Create collections for the ordered sets of sources.
+    ::mstch::array binding_sources(parent_.inputSourcePaths_.begin(),
+                                   parent_.inputSourcePaths_.end());
+
+    // Create a top-level context that contains the extracted information
+    // about this particular binding component.
+    ::mstch::map full_context{{key, context}, {"sources", binding_sources}};
+
+    // Resolve customizable snippets that will be inserted into the file
+    // from the configuration file's "template::file" entry.
+    if (bindingNode_)
+    {
+        chimera::util::extendWithYAMLNode(
+            full_context, bindingNode_["file"], false,
+            std::bind(&chimera::CompiledConfiguration::Lookup, this,
+                      std::placeholders::_1));
+    }
+
+    if (not Render(mangled_name, header_view, "h", context, full_context))
```

Comment:
May we use the `!` here?  I don't know of any new codebases that use `not`, is this becoming popular somewhere?
```suggestion
    if (!Render(mangled_name, header_view, "h", context, full_context))
```

Reply:
I'm using `not` in other codebase and started using it everywhere habitually. :sweat_smile: Let me replace all the use of `not` in chimera to `!`.